wagtailinventory/views.py

```diff
@@ -1,35 +1,52 @@
-from django.core.paginator import Paginator
-from django.http import HttpResponseBadRequest
-from django.shortcuts import render
-from django.views.generic import View
-
+from wagtail.admin.filters import WagtailFilterSet
+from wagtail.admin.views.reports import PageReportView
 from wagtail.models import Page
 
-from wagtailinventory.forms import PageBlockQueryFormSet
-
-
-class SearchView(View):
-    template_name = "wagtailinventory/search.html"
+import django_filters
+from dal import autocomplete
 
-    def get(self, request):
-        if len(request.GET) > 1:
-            formset = PageBlockQueryFormSet(request.GET)
-            if not formset.is_valid():
-                return HttpResponseBadRequest("invalid query")
+from wagtailinventory.models import PageBlock
 
-            queryset = formset.get_query()
-        else:
-            formset = PageBlockQueryFormSet()
-            queryset = Page.objects.all()
 
-        queryset = queryset.order_by("title")
-
-        paginator = Paginator(queryset, per_page=20)
-        pages = paginator.get_page(request.GET.get("p"))
+class BlockAutocompleteView(autocomplete.Select2ListView):
+    def get_list(self):
+        return (
+            PageBlock.objects.distinct()
+            .order_by("block")
+            .values_list("block", flat=True)
+        )
 
-        for page in pages:
-            page.can_choose = True
 
-        return render(
-            request, self.template_name, {"formset": formset, "pages": pages}
-        )
+class BlockInventoryFilterSet(WagtailFilterSet):
+    include_page_blocks = django_filters.AllValuesMultipleFilter(
+        field_name="page_blocks__block",
+        label="Include Blocks",
+        distinct=True,
+        widget=autocomplete.Select2Multiple(
+            url="wagtailinventory:block_autocomplete"
+        ),
+    )
+    exclude_page_blocks = django_filters.AllValuesMultipleFilter(
+        field_name="page_blocks__block",
+        label="Exclude Blocks",
+        distinct=True,
+        exclude=True,
+        widget=autocomplete.Select2Multiple(
+            url="wagtailinventory:block_autocomplete"
+        ),
+    )
+
+    class Meta:
+        model = Page
+        fields = ["include_page_blocks", "exclude_page_blocks"]
+
+
+class BlockInventoryReportView(PageReportView):
+    template_name = "wagtailadmin/reports/base_page_report.html"
```

Comment:
```suggestion
```

I think this line might be unnecessary since it's [in the base class](https://github.com/wagtail/wagtail/blob/682ed3b21fa6924b1e1a6023f80ce7beea570071/wagtail/admin/views/reports/base.py#L46).